libipl/p10/ipl0.C

```diff
@@ -387,9 +388,26 @@ static void process_guard_records()
 		return;
 	}
 
-	openpower::guard::libguard_init(false);
+	openpower::guard::GuardRecords records;
```

Comment:
IMO, we should stop the ipl if the GUARD partition does not exist or is empty because we cannot do much in the error path if wants to create a guard record but, we could continue the ipl for other intermittent error (mostly the standard file operation error will be returned from the libguard_init and getAll function based on the current implementation and its very rare to happen).

It seems now the behavior is the same i.e call the ipl error callback and continue the ipl for any exception from the libguard so your suggestion makes sense to me but, we should consider the libguard::getAll and libguard::clear functions call while adding try and catch for the guard exception and also, we should frame the error message in a generic way for the added ipl error code.

Reply:
Here looks like both cases ipl is continuing right? That seems to be wrong.

Reply:
Addressed the comment from Devender.

Reply:
Discussed and accepted that we will continue to boot for any kind of exception